src/main/java/no/fdk/dataservicecatalog/security/PermissionManager.java

```diff
@@ -35,18 +37,18 @@ public static PermissionManager of(String resourceId, String permission) {
                 .collect(Collectors.toList());
     }
 
-    private String extractCatalogIdFromPath(RequestPath pathContainer) {
-        var path = pathContainer.pathWithinApplication().value();
-        return path.substring(path.indexOf("/catalogs/"), path.indexOf("/dataservices")).replace("/catalogs/", "");
-    }
-
     @Override
     public Mono<AuthorizationDecision> check(Mono<Authentication> mono, AuthorizationContext authorizationContext) {
         return mono.cast(OpenIDAuthenticationToken.class).map(auth -> {
             if (auth.isAuthenticated() && auth.getStatus().equals(OpenIDAuthenticationStatus.SUCCESS)) {
-                var catalogId = extractCatalogIdFromPath(authorizationContext.getExchange().getRequest().getPath());
-                var allowed = getResourceRoles(auth.getAuthorities()).stream().anyMatch(rr -> rr.matchPermission(resourceId, catalogId, permission));
-                return new AuthorizationDecision(allowed);
+                var path = authorizationContext.getExchange().getRequest().getPath().pathWithinApplication().value();
+                if (path.startsWith("/catalogs/")) {
+                    Matcher matcher = catalogIdPattern.matcher(path);
+                    if (matcher.find()) {
```

Comment:
What is the purpose of this check? Don't we already have something similar configured at a higher level?

Reply:
We make sure that the path is actually `/catalogs/{id}/...` and not something like `/some/future/path/with/catalogs/{id}/...`